models/repo_list.go

```diff
@@ -115,76 +115,109 @@ type SearchRepoOptions struct {
 // SearchRepositoryByName takes keyword and part of repository name to search,
 // it returns results in given range and number of total results.
 func SearchRepositoryByName(opts *SearchRepoOptions) (repos RepositoryList, count int64, err error) {
-	var cond = builder.NewCond()
+	// Set Owner ID to Searcher ID if Owner is not filled
+	if opts.OwnerID <= 0 && opts.Searcher != nil {
+		opts.OwnerID = opts.Searcher.ID
+	}
+
+	// Return error if Owner ID is not set
+	if opts.OwnerID <= 0 {
+		return nil, 0, fmt.Errorf("OwnerID is not set")
+	}
+
+	// Check if user with Owner ID exists
+	if opts.Searcher == nil || opts.Searcher.ID != opts.OwnerID {
+		userExists, err := GetUser(&User{ID: opts.OwnerID})
+		if err != nil {
+			return nil, 0, err
+		}
+		if userExists == false {
+			return nil, 0, fmt.Errorf("User with OwnerID doesn't exist")
+		}
+	}
+
+	// Set public search by default
+	isPublicSearch := true
+
+	// Allow to search for owners private data
+	if opts.Searcher != nil && (opts.Searcher.ID == opts.OwnerID || opts.Searcher.IsAdmin) {
+		isPublicSearch = false
+	}
+
+	// Check and set page to correct number
 	if opts.Page <= 0 {
 		opts.Page = 1
 	}
 
-	if opts.Starred && opts.OwnerID > 0 {
+	var cond = builder.NewCond()
+
+	// Include starred repositories by Owner
+	if opts.Starred {
 		cond = builder.Eq{
 			"star.uid": opts.OwnerID,
 		}
 	}
 
-	opts.Keyword = strings.ToLower(opts.Keyword)
+	// Add repository name keyword to search for
 	if opts.Keyword != "" {
+		opts.Keyword = strings.ToLower(opts.Keyword)
 		cond = cond.And(builder.Like{"lower_name", opts.Keyword})
 	}
 
-	// Append conditions
-	if !opts.Starred && opts.OwnerID > 0 {
```

Comment:
`opts.OwnerID > 0` might be a duplicate of https://github.com/Morlinest/gitea/blob/f4729438343b349ea99caa38a4dfef4cefb107d5/models/repo_list.go#L124